examples/multi_agent/tic_tac_toe.py

```diff
@@ -0,0 +1,197 @@
+import os
+import pprint
+from torch.utils.tensorboard import SummaryWriter
+import torch
+import argparse
+import numpy as np
+from typing import Tuple, Optional
+
+from tianshou.env import MAEnv, VectorEnv
+from tianshou.policy import MADQNPolicy
+from tianshou.data import Collector, ReplayBuffer
+from tianshou.trainer import offpolicy_trainer
+
+from MANet import Net
+
+
+class TicTacToeEnv(MAEnv):
+    board = [[0, 0, 0],
+             [0, 0, 0],
+             [0, 0, 0]]
+
+    def __init__(self):
+        super().__init__(None)
+        self.current_board = np.array(TicTacToeEnv.board)
+        self.current_player = 0
+
+    def reset(self) -> dict:
+        self.current_board = np.array(TicTacToeEnv.board)
+        self.current_player = 0
+        return {
+            'agent_id': 0,
+            'obs': np.array(self.current_board),
+            'legal_actions': set(i for i in range(9))
+        }
+
+    def step(self, action: np.ndarray
+             ) -> Tuple[dict, np.ndarray, np.ndarray, dict]:
+        action = action.item(0)
+        assert 0 <= action < 9
+        assert self.current_board.item(action) == 0
+        self._move(action)
+        legal_actions = {i for i, b in enumerate(
+            self.current_board.flatten().tolist()) if b == 0}
```

Comment:
no need of `tolist()` to iterate

Reply:
I want to assure the ``set`` only contains python integers. Not sure how a ``set`` of numpy scalar behaves.

Reply:
> I want to assure the set only contains python integers. 

Yes I got that.

> Not sure how a set of numpy scalar behaves.

It should be the same.

Reply:
Fixed in 279fd28 .